applications/exporter/application_round_applications_exporter.py

```diff
@@ -0,0 +1,271 @@
+    primary_friday: str = ""
+    primary_saturday: str = ""
+    primary_sunday: str = ""
+    secondary_monday: str = ""
+    secondary_tuesday: str = ""
+    secondary_wednesday: str = ""
+    secondary_thursday: str = ""
+    secondary_friday: str = ""
+    secondary_saturday: str = ""
+    secondary_sunday: str = ""
+
+    def add_time_range(self, priority: str, day_of_the_week: str, time_range: SuitableTimeRange) -> None:
+        """Add suitable time range to the correct position in the row."""
+        key = f"{priority.lower()}_{day_of_the_week.lower()}"
+        current_value: str | None = getattr(self, key, None)
+        new_value = f"{local_time_string(time_range.begin_time)}-{local_time_string(time_range.end_time)}"
+        if current_value is None:
+            raise ValueError(f"Key {key} not found in {self.__class__.__name__}")
+        # If the value is already set, append the new time range to the end of the string.
+        # Time ranges should be in chronological order.
+        elif current_value:
+            setattr(self, key, f"{current_value}, {new_value}")
+        else:
+            setattr(self, key, new_value)
+
+
+class ApplicationRoundApplicationsCSVExporter(BaseCSVExporter):
+    """
+    Exports a CSV file containing application data for a single application round.
+
+    Usage:
+    >>> exporter = ApplicationRoundApplicationsCSVExporter(application_round_id=1)
+    >>> csv_export = exporter.export()
+    >>> print(csv_export.getvalue())
+    """
+
+    application_round_id: int
+    max_options: int
+    options_part_template: list[str]
+
+    def __init__(self, application_round_id: int):
+        self.application_round_id = application_round_id
+        self.max_options = 0
+        self.options_part_template = []
+
+    def export(self) -> StringIO | None:
+        application_sections = self._get_queryset()
+        if not application_sections:
+            return None
+
+        self.max_options: int = max(len(section.reservation_unit_options.all()) for section in application_sections)
+        self.options_part_template = [""] * self.max_options
+
+        csv_file = StringIO()
+        csv_writer = csv.writer(csv_file)
+
+        # Write header rows
+        for header_row in self._get_header_rows():
+            csv_writer.writerow(header_row)
+
+        # Write data rows
+        for section in application_sections:
+            csv_writer.writerow(self._get_single_row_data(section))
+
+        return csv_file
+
+    def export_as_file_response(self) -> FileResponse | None:
+        csv_file = self.export()
+        if csv_file is None:
+            return None
+
+        today = local_date()
+        file_name = f"application_round_applications_{self.application_round_id}_{today.isoformat()}.csv"
+        response = FileResponse(csv_file.getvalue(), content_type="text/csv")
+        response["Content-Disposition"] = f"attachment;filename={file_name}"
+        return response
```

Comment:
Should this use the base-class implementation more? Perhaps it can accept a filename and return None if export is None?